From: hololeap <hololeap@protonmail.com>
Signed-off-by: hololeap <hololeap@protonmail.com>
Bug: https://github.com/Avi-D-coder/implicit-hie/pull/40

Add `executable` flag to toggle building the executable

This package is a transient dependency of haskell-language-server.
However, only the library is needed. This flag gives the option for those
who do not need the executable to disable it.

The flag is enabled by default.

diff -urN implicit-hie-0.1.2.5/implicit-hie.cabal implicit-hie-0.1.2.5-new/implicit-hie.cabal
--- implicit-hie-0.1.2.5/implicit-hie.cabal	2001-09-08 19:46:40.000000000 -0600
+++ implicit-hie-0.1.2.5-new/implicit-hie.cabal	2021-01-22 14:34:34.131858938 -0700
@@ -61,7 +61,13 @@
 
   default-language: Haskell2010
 
+flag executable
+  descritption: Build the gen-hie executable
+  default: True
+
 executable gen-hie
+  if !flag(executable)
+    buildable: False
   main-is:          Main.hs
   other-modules:    Paths_implicit_hie
   autogen-modules:  Paths_implicit_hie
